pkgs/applications/audio/mpg123/default.nix

```diff
@@ -1,30 +1,50 @@
-{ lib, stdenv
+{ lib
+, stdenv
 , fetchurl
 , makeWrapper
-, alsa-lib
+, pkg-config
 , perl
-, withConplay ? !stdenv.targetPlatform.isWindows
+, alsa-lib
+, withPulse ? stdenv.hostPlatform.isLinux
+, libpulseaudio
+, AudioUnit
+, AudioToolbox
+, withJack ? stdenv.hostPlatform.isUnix
+, jack
+, withConplay ? !stdenv.hostPlatform.isWindows
 }:
 
 stdenv.mkDerivation rec {
   pname = "mpg123";
-  version = "1.26.5";
+  version = "1.28.2";
 
   src = fetchurl {
     url = "mirror://sourceforge/${pname}/${pname}-${version}.tar.bz2";
-    sha256 = "sha256-UCqX4Nk1vn432YczgCHY8wG641wohPKoPVnEtSRm7wY=";
+    sha256 = "006v44nz4nkpgvxz1k2vbbrfpa2m47hyydscs0wf3iysiyvd9vvy";
   };
 
   outputs = [ "out" ] ++ lib.optionals withConplay [ "conplay" ];
 
-  nativeBuildInputs = lib.optionals withConplay [ makeWrapper ];
+  nativeBuildInputs = lib.optionals withConplay [ makeWrapper ]
+    ++ lib.optionals (withPulse || withJack) [ pkg-config ];
 
   buildInputs = lib.optionals withConplay [ perl ]
-    ++ lib.optionals (!stdenv.isDarwin && !stdenv.targetPlatform.isWindows) [ alsa-lib ];
+    ++ lib.optionals stdenv.hostPlatform.isLinux [ alsa-lib ]
```

Comment:
Personally, I would rather have a `withAlsa ? stdenv.hostPlatform.isLinux` at the top of the file for the inputs

but I guess I don't feel super strongly about this

Reply:
Added `withAlsa` and `withCoreAudio`.